units/concepts.py

```diff
@@ -0,0 +1,112 @@
+from functools import partial
+from itertools import groupby
+
+import structlog
+import httpx
+
+from units.settings import get_settings
+
+logger = structlog.get_logger()
+
+
+def remove_graph_namespaces(s: str) -> str:
+    prefixes = (
+        "http://qudt.org/schema/qudt/",
+        "http://www.w3.org/1999/02/22-rdf-syntax-ns#",
+        "http://www.w3.org/2004/02/skos/core#",
+    )
+
+    for prefix in prefixes:
+        if s.startswith(prefix):
+            return s.replace(prefix, "")
+    return s
+
+
+def language_filter(o: dict, lang: str) -> dict:
+    """Check if given element has the correct language, if one is given"""
+    if "xml:lang" in o and lang not in o["xml:lang"].lower():
+        return False
+    return True
+
+
+def get_qk_for_iri(iri: str) -> str | None:
+    """Get the QUDT quantity key for a given unit IRI."""
+    settings = get_settings()
+    logger.debug("Using sparql endpoint url %s", settings.UNITS_SPARQL_URL)
+
+    QUERY = f"""
+PREFIX qudt: <http://qudt.org/schema/qudt/>
+
+JSON {{
+    "this is ignored anyway": ?qk
+}}
+FROM <{settings.UNITS_VOCAB_PREFIX}qudt/>
+FROM <{settings.UNITS_VOCAB_PREFIX}simapro/>
+WHERE {{
+    <{iri}> qudt:hasQuantityKind ?qk .
+}}
+    """
+
+    logger.debug("Executing query %s", QUERY)
+    response = httpx.post(settings.UNITS_SPARQL_URL, data={"query": QUERY}).json()
+    logger.info(f"Retrieved {len(response)} quantity kind results for iri {iri}")
+    return response[0]["qk"] if response else None
+
+
+def get_all_data_for_qk_iri(
+    iri: str,
+    lang: str | None = None,
+    remove_namespaces: bool = True,
+    graph_namespaces: list[str] = ["qudt", "simapro"],
+) -> dict:
+    """Get all data for a given quantity kind IRI."""
+    settings = get_settings()
+    logger.debug("Using sparql endpoint url %s", settings.UNITS_SPARQL_URL)
+
+    if lang and not len(lang) == 2:
+        raise ValueError("Language code must be exactly two letters long")
```

Comment:
Here we could reuse the same pydantic as for the route and avoid the double validation.